packages/flutter_tools/lib/src/cache.dart

```diff
@@ -1556,7 +1564,11 @@ class ArtifactUpdater {
         _ensureExists(tempFile.parent);
         final IOSink ioSink = tempFile.openWrite();
         await _download(url, ioSink);
+        await ioSink.flush();
         await ioSink.close();
+        if (!tempFile.existsSync()) {
+          throw Exception();
```

Comment:
The exception message will get printed below by the `printTrace()`. You remember what this is for now, but what about in a few  weeks... =)

Reply:
Good point :)